test/CadlRanchProjects/encode/bytes/src/Generated/Header.cs

```diff
@@ -0,0 +1,384 @@
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Encode.Bytes
+{
+    // Data plane generated sub-client.
+    /// <summary> The Header sub-client. </summary>
+    public partial class Header
+    {
+        private readonly HttpPipeline _pipeline;
+        private readonly Uri _endpoint;
+        private readonly string _apiVersion;
+
+        /// <summary> The ClientDiagnostics is used to provide tracing support for the client library. </summary>
+        internal ClientDiagnostics ClientDiagnostics { get; }
+
+        /// <summary> The HTTP pipeline for sending and receiving REST requests and responses. </summary>
+        public virtual HttpPipeline Pipeline => _pipeline;
+
+        /// <summary> Initializes a new instance of Header for mocking. </summary>
+        protected Header()
+        {
+        }
+
+        /// <summary> Initializes a new instance of Header. </summary>
+        /// <param name="clientDiagnostics"> The handler for diagnostic messaging in the client. </param>
+        /// <param name="pipeline"> The HTTP pipeline for sending and receiving REST requests and responses. </param>
+        /// <param name="endpoint"> TestServer endpoint. </param>
+        /// <param name="apiVersion"> The String to use. </param>
+        internal Header(ClientDiagnostics clientDiagnostics, HttpPipeline pipeline, Uri endpoint, string apiVersion)
+        {
+            ClientDiagnostics = clientDiagnostics;
+            _pipeline = pipeline;
+            _endpoint = endpoint;
+            _apiVersion = apiVersion;
+        }
+
+        /// <summary>
+        /// [Protocol Method]
+        /// <list type="bullet">
+        /// <item>
+        /// <description>
+        /// This <see href="https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/ProtocolMethods.md">protocol method</see> allows explicit creation of the request and processing of the response for advanced scenarios.
+        /// </description>
+        /// </item>
+        /// </list>
+        /// </summary>
+        /// <param name="value"> The Bytes to use. </param>
+        /// <param name="context"> The request context, which can override default behaviors of the client pipeline on a per-call basis. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="value"/> is null. </exception>
+        /// <exception cref="RequestFailedException"> Service returned a non-success status code. </exception>
+        /// <returns> The response returned from the service. </returns>
+        /// <include file="Docs/Header.xml" path="doc/members/member[@name='DefaultAsync(BinaryData,RequestContext)']/*" />
+        public virtual async Task<Response> DefaultAsync(BinaryData value, RequestContext context = null)
```

Comment:
I didn't see any convenience API in this client, is it expected?

Reply:
Yes. There is no body model and return model, so it is not needed to generate convenience method.